### PR TITLE
 pb-6810: GetObjLockInfo api implementation for azure in objectstore pkg

### PR DESCRIPTION
**Pull request by siva-portworx**
**What type of PR is this?**
> Uncomment only one and also add the corresponding label in the PR:

>feature

**What this PR does / why we need it**:
```
 pb-6810: GetObjLockInfo api implementation for azure in objectstore pkg
```

**Does this PR change a user-facing CRD or CLI?**:
No

**Is a release note needed?**:
No, We will release note as a feature.

**Does this change need to be cherry-picked to a release branch?**:
No.

Testing:
Vendor the changes to the px-backup and adde
Screenshot 2024-05-09 at 7 28 27 PM
Screenshot 2024-05-09 at 7 42 41 PM
Screenshot 2024-05-09 at 7 42 51 PM
d a Backuplocation with azure immutable bucket with versioning.

### CODE REVIEW

**Review by px-kesavan — APPROVED** [review]
LGTM

**Review by vsundarraj-px — APPROVED** [review]
LGTM

**Review by shkumari-px — APPROVED** [review]
LGTM

File: pkg/objectstore/azure/azure.go [hunk 1]
```diff
@@ -107,6 +109,63 @@ func CreateBucket(backupLocation *stork_api.BackupLocation) error {
...
+			logrus.Errorf("%v: backuplocation: [%v/%v] failed to create client: %v", fn, backupLocation.Namespace, backupLocation.Name, err)
+			return nil, err
+		}
+		res, err := clientFactory.NewBlobContainersClient().GetImmutabilityPolicy(ctx,
+			backupLocation.Location.AzureConfig.ResourceGroupName,
+			backupLocation.Location.AzureConfig.StorageAccountName,
+			backupLocation.Location.Path,
+			&armstorage.BlobContainersClientGetImmutabilityPolicyOptions{IfMatch: nil})
+		if err != nil {
+			logrus.Errorf("%v: failed to  get immutabilty policy for the backuplocation [%v/%v] : %v", fn, backupLocation.Namespace, backupLocation.Name, err)
```
Comment (prashanthpx) [inline pkg/objectstore/azure/azure.go:161]: typo: `immutabilty`

**Review by prashanthpx — APPROVED** [review]
LGTM

File: pkg/objectstore/azure/azure.go (hunk 1 above)
Comment (siva-portworx) [reply pkg/objectstore/azure/azure.go]: Fixed

**Pull request merged**